crates/uv/src/commands/project/run.rs

```diff
@@ -1669,3 +1738,71 @@ fn read_recursion_depth_from_environment_variable() -> anyhow::Result<u32> {
         .parse::<u32>()
         .with_context(|| format!("invalid value for {}", EnvVars::UV_RUN_RECURSION_DEPTH))
 }
+
+#[cfg(unix)]
+fn copy_entrypoint(
+    source: &Path,
+    target: &Path,
+    previous_executable: &Path,
+    python_executable: &Path,
+) -> anyhow::Result<()> {
+    use std::os::unix::fs::PermissionsExt;
+
+    let contents = fs_err::read_to_string(source)?;
+    let expected = r#"#!/bin/sh
+'''exec' "$(dirname -- "$(realpath -- "$0")")"/'python' "$0" "$@"
+' '''
+"#;
```

Comment:
I think we also need to check for absolute shebangs, because we use these in the `--with` environment, but we use standard absolute shebangs in the base environment.

Reply:
We do, that's what https://github.com/astral-sh/uv/blob/d477e368c9c6c7d0c59c1362c11be14846ce7ff1/crates/uv/src/commands/project/run.rs#L1760 does (and there's test coverage for both)

Reply:
I updated it to be clearer